connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java

```diff
@@ -106,6 +108,17 @@ public void stop() {
         Utils.closeQuietly(targetAdminClient, "target admin client");
     }
 
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        List<ConfigValue> configValues = super.validate(connectorConfigs).configValues();
+        new MirrorCheckpointConfig(connectorConfigs).validate().entrySet().forEach(invalidConfig ->
```

Comment:
Ah, I just remembered--I don't think we want to do `new MirrorCheckpointConfig(...)` in this method, since it'll throw an exception if there are any problems with the config that get detected during that phase (such as an integer property being given a value of `"notanint"`). @mimaison raised the same concern on an earlier PR of mine that added logic to `MirrorSourceConnector::validate` [here](https://github.com/apache/kafka/pull/12366#discussion_r1101366791).

Can we replicate this logic without instantiating a `MirrorCheckpointConfig` object?

Also, small nit--doing `map.entrySet().forEach(...)` is redundant; you can just do `map.forEach((key, value) -> ...)`.

Reply:
ah this actually make sense! The PR was originally following the same approach as `MirrorSourceConnector::validate`. I changed `MirrorCheckpointConfig::validate` to static and dropped the initialisation of `MirrorCheckpointConfig`. I still think config classes should have the validation logic within whenever it is possible.

Reply:
Fully agree that the validation logic makes more sense in the config class; this looks great. Thanks for the update!